### PR TITLE
[TECHNICAL SUPPORT] LPS-107796 Order and labels of fields should be as last form version

### PR DESCRIPTION
**Pull request by rodrigopaulino**
Relevant ticket:
https://issues.liferay.com/browse/LPS-107796

### CODE REVIEW

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by rodrigopaulino** [comment]
ci:test:relevant

**Comment by natocesarrego** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordExporterImpl.java [hunk 1]
```diff
@@ -276,6 +281,12 @@ protected String getStatusMessage(int status, Locale locale) {
 			ddmFormInstanceVersionLocalService.getFormInstanceVersions(
 				ddmFormInstanceId, QueryUtil.ALL_POS, QueryUtil.ALL_POS, null);
 
+		ddmFormInstanceVersions = ListUtil.copy(ddmFormInstanceVersions);
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordExporterImpl.java]: Why did you need to create this copy before sort the versions, @rodrigopaulino?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordExporterImpl.java [hunk 2]
```diff
@@ -249,7 +252,9 @@ protected String getDDMFormFieldValue(
 		stream.map(
 			this::getNontransientDDMFormFieldsMap
 		).forEach(
-			ddmFormFields::putAll
+			map -> map.forEach(
+				(key, ddmFormField) -> ddmFormFields.putIfAbsent(
+					key, ddmFormField))
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordExporterImpl.java]: @rodrigopaulino, can you please clarify why this change was necessary? Which problem is it exactly addressing?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordExporterImpl.java (hunk 1 above)
Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordExporterImpl.java]: Honestly, it's probably here because I picked this hunk of code from the 7.1.x branch where the sort of versions is correct. But I don't see why it's necessary either. I'll make sure to remove it from 7.1.x when backporting the fix.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordExporterImpl.java (hunk 2 above)
Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordExporterImpl.java]: This is the verification we've talked offline that is necessary to maintain the latest version field labels. If we simply put the same field in that order, the label exported would be the first version's, and that is the original bug reported on 7.1.x.

Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/exporter/DDMFormInstanceRecordExporterImpl.java]: Makes sense to me! Thanks for the clarification, @rodrigopaulino. 😉

**Comment by natocesarrego** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/84305. [See changes here.](https://github.com/natocesarrego/liferay-portal/compare/80210c234c48~1...80210c234c48) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by natocesarrego** [comment]
Pull request submitted to https://github.com/natocesarrego/liferay-portal/pull/281. [See changes here.](https://github.com/natocesarrego/liferay-portal/compare/80210c234c48~1...80210c234c48) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*
